core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -1952,3 +2012,294 @@ oppia.factory('explorationWarningsService', [
+          var stateName = lostChange.state_name;
+          if (!stateWiseEditsMapping[stateName]) {
+            stateWiseEditsMapping[stateName] = [];
+          }
+
+          switch (lostChange.property_name) {
+            case 'content':
+              if (newValue !== null) {
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div></div>').html(
+                    'Edited content: ' + newValue.value));
+              }
+              break;
+
+            case 'widget_id':
+              if (oldValue === null) {
+                if (newValue !== 'EndExploration') {
+                  lostChangeValue = 'Added Interaction: ' + newValue;
+                } else {
+                  lostChangeValue = 'Ended Exploration';
+                }
+              } else {
+                lostChangeValue = 'Deleted Interaction: ' + oldValue;
+              }
+              stateWiseEditsMapping[stateName].push(
+                angular.element('<div></div>').html(lostChangeValue));
+              break;
+
+            case 'widget_customization_args':
+              if (utilsService.isEmpty(oldValue)) {
+                lostChangeValue = 'Added Interaction Customizations';
+              } else if (utilsService.isEmpty(newValue)) {
+                lostChangeValue = 'Removed Interaction Customizations';
+              } else {
+                lostChangeValue = 'Edited Interaction Customizations';
+              }
+              stateWiseEditsMapping[stateName].push(
+                angular.element('<div></div>').html(lostChangeValue));
+              break;
+
+            case 'answer_groups':
+              var answerGroupChanges = getRelativeChangeToGroups(lostChange);
+              var answerGroupHtml = '';
+              if (answerGroupChanges === 'added') {
+                answerGroupHtml += (
+                  '<p>Destination: ' + newValue.outcome.dest + '</p>');
+                answerGroupHtml += (
+                  '<p>Feedback:' + newValue.outcome.feedback + '</p>');
+                var rulesList = makeRulesListHumanReadable(newValue);
+                if (rulesList.length) {
+                  answerGroupHtml += ('<div>Rules: <div>');
+                  for (var rule in rulesList) {
+                    answerGroupHtml += rulesList[rule].html();
+                  }
+                }
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div>Added answer group: </div>')
+                    .html(answerGroupHtml));
+              } else if (answerGroupChanges === 'edited') {
+                if (newValue.outcome.dest !== oldValue.outcome.dest) {
+                  answerGroupHtml += (
+                    '<p>Destination: ' + newValue.outcome.dest + '</p>');
+                }
+                if (!angular.equals(
+                  newValue.outcome.feedback, oldValue.outcome.feedback)) {
```

Comment:
This line should be indented by four spaces more than the preceeding line, as it is an if overflow.